app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java

```diff
@@ -0,0 +1,186 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.R;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.EventListener;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+
+/**
+ * Making a dialogfragment from fragment came from the following video:
+ * https://www.youtube.com/watch?v=LUV_djRHSEY
+ *
+ * Name: Mitch Tabian
+ * Video Date: December 10, 2017
+ */
+public class AddHabitDialog extends DialogFragment{
+
+    private static final String TAG = "AddHabitDialog";
+    FirebaseFirestore db;
+
+    public  interface HabitDialogListener {
+        void addNewHabit(String title, String reason, Date date);
+    }
```

Comment:
Can this interface be abstracted to it's **own class**? If so, that's what we should do that here instead of having it within the class itself. Then, this class can **implement** the interface.